doc/reference/actor.md

```diff
@@ -2,755 +2,734 @@
 ```
-:::
 
-Please document me!
+Stops the remote server with identifier `srv-id`.
 
-### rpc.disconnect
-::: tip usage
+### remote-list-actors
+```scheme
+(remote-list-actors srv-id (srv (current-actor-server)))
 ```
-(rpc.disconnect ...)
-(!rpc.disconnect ...)
-(!!rpc.disconnect ...)
-```
-:::
 
-Please document me!
+Lists registered actors at the remote server with identifier `srv-id`.
 
-### rpc.shutdown
-::: tip usage
+### remote-connect-to-server!
+```scheme
+(remote-connect-to-server! from-id to-id (addrs #f) (srv (current-actor-server)))
 ```
-(rpc.shutdown ...)
-(!rpc.shutdown ...)
-(!!rpc.shutdown ...)
-```
-:::
 
-Please document me!
+Asks the remote server with identifier `from-id` to connect to the
+server with identifier `to-id`.  If the optional addresses `addrs` are
+not specified, then the `to-id` server will be looked up in the registry.
 
+### remote-list-connections
+```scheme
+(remote-list-connections srv-id (srv (current-actor-server)))
+```
 
-## RPC Wire Protocols
+Lists the connections of a remote server with identifier `srv-id`.
 
-### rpc-null-proto
-::: tip usage
+### remote-load-library-module
+```scheme
+(remote-load-library-module srv-id mod (srv (current-actor-server)))
 ```
-(rpc-null-proto ...)
-```
-:::
 
-Please document me!
+Asks the remote server with identifier `srv-id` to load the library module `mod`.
 
-### rpc-cookie-proto
-::: tip usage
-```
-(rpc-cookie-proto ...)
+### remote-load-code
+```scheme
+(remote-load-code srv-id object-file-path (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Asks the remote server with identifier `srv-id` to load a code object file.
 
-### rpc-generate-cookie!
-::: tip usage
-```
-(rpc-generate-cookie! ...)
+### remote-eval
+```scheme
+(remote-eval srv-id expr (srv (current-actor-server)))
 ```
-:::
 
-Please document me!
+Evaluates `expr` in the remote server with identifier `srv-id`.
 
-### rpc-cipher-proto
-::: tip usage
+### ping-server
+```scheme
+(ping-server srv-id (srv (current-actor-server)))
 ```
-(rpc-cipher-proto ...)
-```
-:::
 
-Please document me!
+Pings the remote server with identifier `srv-id`.
 
-### rpc-cookie-cipher-proto
-::: tip usage
+### ping-actor
+```scheme
+(ping-actor ref (srv (current-actor-server)))
 ```
-(rpc-cookie-cipher-proto ...)
-```
-:::
 
-Please document me!
+Pings the actor referred by `ref`.
 
+### ensemble-add-server!
+```scheme
+(ensemble-add-server! id addrs roles (srv (current-actor-server)))
+```
 
-## RPC Options
+Adds a server to the ensemble registry:
+- `id` is the server identifier
+- `addrs` is the list of actor server addresses.
+- `roles` is a list of roles for the server; a list of symbols.
 
-### set-rpc-keep-alive-interval!
-::: tip usage
+### ensemble-remove-server!
+```scheme
+(ensemble-remove-server! id (srv (current-actor-server)))
 ```
-(set-rpc-keep-alive-interval! ...)
-```
-:::
 
-Please document me!
+Removes the server with identifier `id` from the ensemble registry.
```

Comment:
Should these procedures be available only to a specific "control" server? Seems a bit weird to allow anyone to issue such commands.

Maybe add some capabilities slot/flag to each server in the registry. For command and control operations (or `eval` as mentioned above), require the appropriate flag.

Have a special handshake if you want to upgrade the connection to C&C. Not sure how to (dis)allow operations cost-efficiently though.

Reply:
We _could_ require some sort of admin priviledges for certain operations.

It will complicate the API somewhat, but maybe it is a good idea in the long term.
I am inclined to leave it as is for now -- let's revisit this in near future.

Reply:
It's not clear how to do this without runtime checks.

Reply:
#766 has a pretty clean solution with minimally intrusive api.